internal/muxado/buffer.go

```diff
@@ -68,7 +113,12 @@ func (b *inboundBuffer) Read(p []byte) (n int, err error) {
 			err = b.err
 			break
 		}
+
+		b.startTimerLocked(timeout)
```

Comment:
isn't this going to start the timer in all cases (even when we don't have a deadline), but also use `0` as a timeout, effectively waking up the goroutine all the time? And then continue to set/wake at a rapid pace?

Reply:
Ah! I think this is the bug that's causing my test failures. I was also shadowing rather than setting the `timeout` variable. Thanks Go.